toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -540,48 +759,64 @@ func (im *IsoMaker) buildIsoImageFilePath() string {
 
 // deferIsoMakerCleanUp accepts clean-up tasks to be ran when the entire
 // build process has finished, NOT at the end of the current scope.
-func (im *IsoMaker) deferIsoMakerCleanUp(cleanUpTask func()) {
+func (im *IsoMaker) deferIsoMakerCleanUp(cleanUpTask func()(error)) {
 	im.isoMakerCleanUpTasks = append(im.isoMakerCleanUpTasks, cleanUpTask)
 }
 
 // isoMakerCleanUp runs all clean-up tasks scheduled through "deferIsoMakerCleanUp".
 // Tasks are ran in reverse order to how they were scheduled.
-func (im *IsoMaker) isoMakerCleanUp() {
+func (im *IsoMaker) isoMakerCleanUp() (error err) {
 	// We defer again to run the tasks in the correct order AND so that a potential
 	// panic from one of these doesn't prevent other ones from running.
 	for _, cleanUpTask := range im.isoMakerCleanUpTasks {
-		defer cleanUpTask()
+		defer func() {
+			tmpErr := cleanUpTask()
+			if tmpErr != nil {
+				if err != nil {
+					err = fmt.ErrorF("Error:%v\nError:%v\n", err, tmpErr) 
+				} else {
+					err = fmt.ErrorF("Error:%v\n", tmpErr) 
+				}
+			}
+		}
 	}
 }
 
-func readConfigFile(configFilePath, baseDirPath string) configuration.Config {
+func readConfigFile(configFilePath, baseDirPath string) (configuration.Config, error) {
 	config, err := configuration.LoadWithAbsolutePaths(configFilePath, baseDirPath)
-	logger.PanicOnError(err, "Failed while reading config file from '%s' with base directory '%s'.", configFilePath, baseDirPath)
-	return config
+	if err != nil {
+		return configuration.Config{}, fmt.Errorf("Failed while reading config file from '%s' with base directory '%s'.\nError:%v", configFilePath, baseDirPath, err)
```

Comment:
nit: In Go, errors start with a lower case letter.

Reply:
fixed.